R/rendering.R

```diff
@@ -336,9 +336,79 @@ getCurveIDAnalsysiGroupResults <- function(curveids, ...) {
 #' curveData <- poIVPKCurveData$points
 #' plotCurve(curveData, params, paramNames = NA, outFile = NA, ymin = NA, logDose = FALSE, logResponse=TRUE, ymax = NA, xmin = NA, xmax = NA, height = 300, width = 300, showGrid = FALSE, showLegend = FALSE, showAxes = TRUE, plotMeans = FALSE, connectPoints = TRUE, drawCurve = FALSE, addShapes = TRUE, drawStdDevs = TRUE)
 #' 
+filterFlaggedPoints <- function(points, returnGood = TRUE) {
+  badFlag <- "knocked out"
+  if(returnGood) {
+    return(subset(points, userFlagStatus!=badFlag & preprocessFlagStatus!=badFlag & algorithmFlagStatus!=badFlag & tempFlagStatus!=badFlag))
+  } else {
+    return(subset(points, userFlagStatus==badFlag | preprocessFlagStatus==badFlag | algorithmFlagStatus==badFlag | tempFlagStatus==badFlag))
+  }
+}
+applyParsedParametersToFitData <- function(fitData, parsedParams, protocolDisplayValues) {
+    ## PREP FROM RENDER CURVE
+    # Colors
+    # plotColors <- c("black", "#0C5BB0FF", "#EE0011FF", "#15983DFF", "#EC579AFF", "#FA6B09FF", 
+    #                 "#149BEDFF", "#A1C720FF", "#FEC10BFF", "#16A08CFF", "#9A703EFF")
+    if(!is.null(racas::applicationSettings$server.curveRender.plotColors)) {
+        plotColors <- trimws(strsplit(racas::applicationSettings$server.curveRender.plotColors,",")[[1]])
+    } else {
+        plotColors <- c("black", "#0C5BB0FF", "#EE0011FF", "#15983DFF", "#EC579AFF", "#FA6B09FF", 
+                        "#149BEDFF", "#A1C720FF", "#FEC10BFF", "#16A08CFF", "#9A703EFF")
+    }
+    if(!is.na(parsedParams$colorBy)) {
+        key <- switch(parsedParams$colorBy,
+                    "protocol" = "protocol_label",
+                    "experiment" = "experiment_label",
+                    "batch" = "batch_code"
+        )
+        uniqueKeys <- setkeyv(unique(fitData[ , key, with = FALSE]), key)
+        colorCategories <- uniqueKeys[, color:=rep(plotColors, length.out = .N)][ , name:=get(key)]
+        fitData <- merge(fitData, colorCategories, by = key)
+    }
 
-plotCurve <- function(curveData, params, outFile = NA, ymin = NA, logDose = FALSE, logResponse = FALSE, ymax = NA, xmin = NA, xmax = NA, height = 300, width = 300, showGrid = FALSE, showLegend = FALSE, showAxes = TRUE, drawCurve = TRUE, drawFlagged = FALSE, plotMeans = FALSE, drawStdDevs = FALSE, addShapes = FALSE, labelAxes = FALSE, curveXrn = c(NA, NA), mostRecentCurveColor = NA, axes = c("x","y"), modZero = TRUE, drawPointsForRejectedCurve = racas::applicationSettings$server.curveRender.drawPointsForRejectedCurve, plotColors = c("black"),curveLwd = 1, plotPoints = TRUE, xlabel = NA, ylabel = NA) {
+    if("category" %in% names(fitData)) {
+      fitData <- fitData[exists("category") & (!is.null(category) & category %in% c("inactive","potent")), c("fittedMax", "fittedMin") := {
+        responseMean <- mean(points[[1]][userFlagStatus!="knocked out" & preprocessFlagStatus!="knocked out" & algorithmFlagStatus!="knocked out" & tempFlagStatus!="knocked out",]$response)
```

Comment:
Some of this is really dense and it's hard for me to follow what this is doing, and why it's doing it. This function appears to have a bunch of important logic about built-in default behaviors. I'd appreciate a lot more comments throughout this new function to describe those behaviors.